src/libraries/Microsoft.Extensions.Logging/tests/Microsoft.Extensions.Logging.Generators.Tests/LoggerMessageGeneratorEmitterTests.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using SourceGenerators.Tests;
+using Xunit;
+
+namespace Microsoft.Extensions.Logging.Generators.Tests
+{
+    [ActiveIssue("https://github.com/dotnet/runtime/issues/32743", TestRuntimes.Mono)]
+    public class LoggerMessageGeneratorEmitterTests
+    {
+        [Fact]
+        public async Task TestEmitter()
+        {
+            // The functionality of the resulting code is tested via LoggerMessageGeneratedCodeTests.cs
+            string[] sources = Directory.GetFiles("TestClasses");
+            foreach (var src in sources)
+            {
+                var testSourceCode = await File.ReadAllTextAsync(src).ConfigureAwait(false);
+
+                var (d, r) = await RoslynTestUtils.RunGenerator(
+                    new LoggerMessageGenerator(),
+                    new[] { Assembly.GetAssembly(typeof(ILogger))!, Assembly.GetAssembly(typeof(LoggerMessageAttribute))! },
+                    new[] { testSourceCode }).ConfigureAwait(false);
+
+                Assert.Empty(d);
+                _ = Assert.Single(r);
```

Comment:
We are just asserting that the generator didn't fail? Should we be baselining the output or something?

Reply:
- [ ] Add a set of baseline files to test against (two or three use cases)

Reply:
I'm not sure there needs to be any more tests here. The model is this:

* Everything in TestClasses/*.cs is processed by the code generator. In order for the test suite to compile, the code generators need to at least work to some level of quality.

* The code that was thus generated is then tested by explicit tests in the GeneratedCode tests. This ensures that the code produced by the code generator behaves as expected.

* The emitter tests exists primarily to replicate what the generator has already done against TestClasses/*. The difference is that this time the generator is run from within the test infra so it's possible to collect coverage.

* The explicit parser tests are there primarily to exercise the error paths in the parser, which are necessarily not triggered by the code in TestClasses/*.cs (otherwise they wouldn't compile).

The one potential gap here is that there is no assurance that the code produced by the emitter tests actually matches the code produced by actually running the generator when building the test suite. So it might be good to compare the generated .cs file in the test suite's obj/ directory vs. what the emitter tests produce. These should match byte-for-byte.

Reply:
One benefit of also using baselines is that you can review exactly how a change to the generator will affect the generated code. Code reviewers can see exactly what code is generated, and what was previously generated before the change.